Packs/AzureDevOps/Integrations/AzureDevOps/AzureDevOps_test.py

```diff
@@ -0,0 +1,748 @@
+    with pytest.raises(Exception):
+        pipeline_list_command(client, {"project": project,
+                                       'limit': '-1'})
+
+
+def test_azure_devops_branch_list_command(requests_mock):
+    """
+    Scenario: Retrieve repository branches list.
+    Given:
+     - User has provided valid credentials.
+    When:
+     - azure-devops-branch-list called.
+    Then:
+     - Ensure number of items is correct.
+     - Ensure outputs prefix is correct.
+     - Ensure a sample value from the API matches what is generated in the context.
+    """
+    from AzureDevOps import Client, branch_list_command
+
+    authorization_url = 'https://login.microsoftonline.com/organizations/oauth2/v2.0/token'
+    requests_mock.post(authorization_url, json=get_azure_access_token_mock())
+
+    project = 'xsoar'
+    repository = 'test'
+
+    url = f'{BASE_URL}/{project}/_apis/git/repositories/{repository}/refs'
+
+    mock_response = json.loads(load_mock_response('branch_list.json'))
+    requests_mock.get(url, json=mock_response)
+
+    client = Client(
+        client_id=CLIENT_ID,
+        organization=ORGANIZATION,
+        verify=False,
+        proxy=False)
+
+    result = branch_list_command(client, {"project": project, "repository": repository})
+
+    assert len(result.outputs) == 2
+    assert result.outputs_prefix == 'AzureDevOps.Branch'
+    assert result.outputs[0].get('repository') == repository
+    assert result.outputs[0].get('name') == 'refs/heads/main'
+    with pytest.raises(Exception):
+        branch_list_command(client, {"project": project,
+                                     "repository": repository,
+                                     'limit': '-1'})
+
+
+def test_get_last_fetch_incident_index(requests_mock):
+    """
+    Scenario: Retrieve the index of the last fetched pull-request.
+    Given:
+     - User has provided valid credentials.
+    When:
+     - fetch-incidents called.
+    Then:
+     - Ensure the return value is correct.
+    """
+    from AzureDevOps import Client, get_last_fetch_incident_index
+
+    authorization_url = 'https://login.microsoftonline.com/organizations/oauth2/v2.0/token'
+    requests_mock.post(authorization_url, json=get_azure_access_token_mock())
+
+    project = 'test'
+    repository = 'xsoar'
+
+    url = f'{BASE_URL}/{project}/_apis/git/repositories/{repository}/pullrequests/'
+
+    mock_response = json.loads(load_mock_response('list_pull_request.json'))
+    requests_mock.get(url, json=mock_response)
+
+    client = Client(
+        client_id=CLIENT_ID,
+        organization=ORGANIZATION,
+        verify=False,
+        proxy=False)
+
+    assert get_last_fetch_incident_index(project, repository, client, 24) == 0
+    assert get_last_fetch_incident_index(project, repository, client, 22) == 1
+    assert get_last_fetch_incident_index(project, repository, client, 25) == -1
+    assert get_last_fetch_incident_index(project, repository, client, 21) == -1
```

Comment:
This is usually done with `parametrize` pytest prefix. See Netscout Arbor Sightline for example.